### ISSUE TITLE
Might be a Creative to Survival Bug

### ISSUE THREAD

**Issue by CampingSteve** [Rocologo/BagOfGold#13]
When you have a bag of Gold in your survival inv (not in the slots you can access with numbers 1 - 9)  and you go into creative, then when you go back into survival you get a Bag of Golds Head instead of your BagofGold with money in it.

**Comment by Rocologo** [comment]
Ok, I will test to see if I can reproduce this. :-)

**Comment by CampingSteve** [comment]
I think you just have to save the players current inv balance on gamemode change from survival to creative

**Comment by CampingSteve** [comment]
hmmm just tried it again and this time it duplicated my bag in my inv lol

**Comment by Rocologo** [comment]
The problem is not that simple :-) and I save the balance all the time. I also need to make sure what happens, if PerWorldInventory is used and not.

I got a dublication too... except one bag had a value and the other did not. So balance was still equal the bagofgold value.

**Comment by CampingSteve** [comment]
if you have a bagofgold in the hidden inv in creative it duplicates it on return to survival

**Comment by CampingSteve** [comment]
i'm not using perworld

**Comment by CampingSteve** [comment]
it seems not to happen everytime too

**Comment by CampingSteve** [comment]
tricky bug

**Comment by Rocologo** [comment]
confirmed. but the item in the hidden part has no value. (The display name Is  correct, but the hidden data is missing) so it has no value.

This is when you use PWI like I do :-) 

I will test without PWI to see what happens then :-)

**Comment by CampingSteve** [comment]
Think I found how I get the bagofgold heads....I put the duplicated bags of gold on the floor, then picked them back up and they turn into Bagofgold head

**Comment by Rocologo** [comment]
I will of cause try to fix this, but its not a big problem. Because you can always get copies when you allow uses to use creative. 

Example
1) go to creative 
2) place a bagofgold on the ground
3) go to survival
4) break the bagofgold and the player has a copy (without value) and I can do anything about it.

**Comment by Rocologo** [comment]
If you disable ProtocolLb for a short while you will be able to see whats going on.

The value is removed before you place the bag on the ground.

**Comment by CampingSteve** [comment]
its not a big bug...99% of players wont notice. I just want you to be aware of any issues that's all ;)

**Comment by Rocologo** [comment]
And I appreciate this!!!

I will try to see if I can remove the duplicate.

**Comment by CampingSteve** [comment]
I  think the bagofgold was highlighted too as a selection

**Comment by CampingSteve** [comment]
tbh haven't got a clue lmfao

**Comment by Rocologo** [comment]
If you allow your player to change between survival and creative the will ALWAYS be able copy/move items from creative to survival. No matter what, they will always be able to go to creative, place the blocks on gound the want in survival, go to survival and then break the blocks.

This is why all the worlds on my server is purely survival or creative, and players is not allowed to change gamemode

**Comment by CampingSteve** [comment]
same here...it just happened when I was making something at spawn

**Comment by Rocologo** [comment]
I have looked at the problem, and I can solve the problem if the player has only one bag in the hidden inventory.

But if the starting balance is 1000 in three bags and you change gamemode from survival to creative and back I end up with one bag with 1000 and two bags with 0. This is because I only save the balance and not the bags (and how many money they contains)....

Anyway - This is not a big problem, because the players are not able to earn money this way. The balance is kept steady and this is whats most important! 

Would it be okay if I make the solution this way:

1) Starting balance 1000 in three bags
2) GameMode changes
3) Ending balance 1000 in ONE bag

Would that be okay?

**Comment by CampingSteve** [comment]
Yes that's great...doesn't really matter how many bags you have 😁

**Comment by Rocologo** [comment]
I'll make a new release when you say go :-)

**Comment by CampingSteve** [comment]
I can't test till later as I'm going out this afternoon

**Comment by CampingSteve** [comment]
downloaded them will test after dinner ;)

**Comment by CampingSteve** [comment]
still gives the heads but the value of the bags is more stable...and balance isn't affected

**Comment by CampingSteve** [comment]
found another one, where you put a bag of gold into a shulker box and it sometimes gives a bagofgold head when its taken out

**Comment by Rocologo** [comment]
I dont get ekstra heads. They are removed when i try????

I will investigate the shulker box, i thought i had disabled all non real inventories. But i will check again.

**Comment by CampingSteve** [comment]
do I have to remove the previous directories?

**Comment by CampingSteve** [comment]
and do a clean install?

**Comment by Rocologo** [comment]
No, you dont need to delete anything

**Comment by CampingSteve** [comment]
ok

**Comment by CampingSteve** [comment]
Added to my server to test, i'll put on /mh debug and could you join to test as a new player?
136.243.179.186:60202

**Comment by Rocologo** [comment]
Please test 1.1.0 + 5.6.1

**Comment by CampingSteve** [comment]
I will when I get back from holiday

**Comment by Rocologo** [comment]
Hi @CampingSteve do you still use BagOfGold I have not got any new bug reports after i released V1.1.3. Could you please help me with the testing?

**Comment by Rocologo** [comment]
including MobHunting v5.7.0

**Comment by Rocologo** [comment]
Please test on newest 1.1.4-SNAPSHOT
https://fractal.lindegaard.one:8181/job/BagOfGold/

**Comment by Rocologo** [comment]
And 5.7.1-SNAPSHOT
https://fractal.lindegaard.one:8181/job/MobHunting/

**Comment by Rocologo** [comment]
I have moved to MC 1.13 now. Where I have added REAL Multiworld support

**Comment by CampingSteve** [comment]
Yes mate...i still use bag of gold its running great so far not noticed any bugs yet....will try and break it tonight when i get in after work lol

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
